src/s_macos.mm

```diff
@@ -0,0 +1,52 @@
+#import <Cocoa/Cocoa.h>
+
+static volatile int sys_eventloop_running;
+
+extern "C" void sys_eventloop_setup(void)
+{
+    [NSApplication sharedApplication];
+}
+
+extern "C" void sys_eventloop_run(void)
```

Comment:
Maybe it's good to include some commenting as why this is here and what the code is for.